packages/teleport-project-packer/src/index.ts

```diff
@@ -49,37 +47,24 @@ export const createProjectPacker: PackerFactory = (params: PackerFactoryParams =
     assets = assetsToSet
   }
 
-  const setTemplate = (templateToSet: TemplateDefinition): void => {
-    template = templateToSet
+  const setTemplate = (templateFolder: GeneratedFolder): void => {
+    template = templateFolder
   }
 
   const loadTemplate = async (
-    templateToLoad?: TemplateDefinition
-  ): Promise<LoadTemplateResponse> => {
-    template = templateToLoad || template || DEFAULT_TEMPLATE
-
-    if (template.templateFolder) {
-      return { success: true, payload: template.templateFolder }
-    }
-
-    if (!template.remote) {
-      return { success: false, payload: NO_REMOTE_TEMPLATE_PROVIDED }
-    }
-
+    remoteDefinition: RemoteTemplateDefinition
+  ): Promise<GeneratedFolder> => {
     try {
-      template.templateFolder = await fetchTemplate(template.remote)
-      templateLoaded = true
-      return { success: true, payload: template.templateFolder }
+      template = await fetchTemplate(remoteDefinition)
+      return template // TODO: is this useful?
```

Comment:
There might be cases in which, after the user loads the template, wants to parse it in some way. So I think that returning the template after we are loading it is a good thing. Maybe just:

```js
return await fetchTemplate(remoteDefinition)
```

Reply:
but then the template inside the packer closure would not be updated. we would have to call setTemplate after loadTemplate